sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/EliminateJoinToEmptyRelation.scala

```diff
@@ -33,6 +34,9 @@ import org.apache.spark.sql.execution.joins.{EmptyHashedRelation, HashedRelation
  *    This applies to all Joins (sort merge join, shuffled hash join, and broadcast hash join),
  *    because sort merge join and shuffled hash join will be changed to broadcast hash join with AQE
  *    at the first place.
+ *
+ * 3. Join is left anti join without condition, and broadcasted join right side is not empty.
+ *    This applies to broadcast nested loop join only.
```

Comment:
Why do we care about the physical plan? I think it's a logical optimization that left anti join without condition can be turned into empty relation if right side is non-empty.

Reply:
@cloud-fan - I agree. Updated to remove this restriction. I was not thinking towards to use row count stats at the first place, so was checking the size of `Array[InternalRow]` for `BroadcastNestedLoopJoinExec`, so was having this restriction.